packages/js/src/decorator/tinyMCE.js

```diff
@@ -36,11 +35,29 @@ function markTinyMCE( editor, paper, marks ) {
 	let html = editor.getContent();
 	html = markers.removeMarks( html );
 
+	/*
+	 * Get the information whether we want to mark a specific part of the HTML. If we do, `fieldsToMark` should return an array with that information.
+	 * For example, [ "subehading" ] means that we want to apply the markings in subheadings only, and not the other parts.
+	 * `selectedHTML` is an array of the HTML parts that we want to apply the marking to.
+	 */
 	const { fieldsToMark, selectedHTML } = languageProcessing.getFieldsToMark( marks, html );
-
 	// Generate marked HTML.
 	forEach( marks, function( mark ) {
+		/*
+		 * Classic editor uses double quotes for HTML attribute values. However, Block editor uses single quotes for HTML tag attributes,
+		 * and that's why in `yoastseo`, we use single quotes for the attribute values when we create the marked object. As a result,
+		 * the replacement did not work, as the marks passed by `yoastseo` did not match anything in the original text.
+		 * This step is replacing the single quotes in the marked object output by `yoastseo` with double quotes.
+		 * This way, we make sure that the replacement can find a match between the original text of the marked object and the text in the page.
+		 */
+		if ( ! editor.id === "acf_content" ) {
```

Comment:
This checks whether Advanced Custom Fields is active.

Reply:
> However, what about the case when the ACF plugin is activated together with Classic editor? this means that we wouldn't replace single quotes with double quotes?

@hdvos What happens in that case?

Reply:
In that case the highlighting does not work.